lumicks/pylake/kymotracker/detail/velocity.py

```diff
@@ -0,0 +1,254 @@
+
+    return PiecewiseModel(
+        intercept, [slope], [], intercept_std, [slope_std], [], rss, bic, "converged"
+    )
+
+
+def _optimize_breakpoints(x, y, n_breakpoints, tol, max_iter):
+    r"""
+    Optimization of breakpoint locations for a continuous piecewise linear model
+
+    Parameters
+    ----------
+    x : np.ndarray
+        Dependent variable (time).
+    y : np.ndarray
+        Independent variable (position).
+    n_breakpoints : int
+        Number of requested breakpoints.
+    tol : float
+        Convergence tolerance. Ignored if `n_breakpoints == 0`.
+    max_iter : int
+        Maximum number of iterations allowed in the optimization algorithm. Ignored if
+        `n_breakpoints == 0`.
+    n_restarts : int
+        Number of times to run the optimization algorithm with breakpoint initial guesses drawn
+        from a uniform distribution. Ignored if `n_breakpoints == 0`.
+    """
+    n_samples = len(x)
+    n_coeffs = 2 + 2 * n_breakpoints
+
+    def fit_iteration(breakpoints):
+        r"""
+        Single iteration of the breakpoint estimation algorithm, solving Eq 7. corresponding
+        to linearized form of piecewise continuous model, with nonlinear breakpoints term:
+
+        .. math::
+
+            \alpha x + \beta U + \gamma V
+
+        where :math:`\alpha` is the slope of the first section, :math:`\beta` models the
+        difference-in-slopes of the subsequent sections, and :math:`\gamma` is a reparameterization
+        of the breakpoints :math:`\psi`.
+
+        Given the current breakpoints guess :math:`\psi^{(0)}` and the ML estimates of above
+        parameters using OLS, the updated breakpoints can be calculated as:
+
+        .. math::
+
+            \psi = \frac{\gamma}{\beta} + \psi^{(0)}
+        """
+        diff_term = x - breakpoints[:, np.newaxis]
+        heavi = np.vstack([np.heaviside(d, -1) for d in diff_term])
+        u = diff_term * heavi
+        v = -heavi
+        design_matrix = np.vstack((np.ones(x.size), x, u, v)).T
+
+        xtx = np.linalg.pinv(np.matmul(design_matrix.T, design_matrix))
+        coeffs = np.dot(
+            xtx,
+            np.dot(design_matrix.T, y),
+        )
+
+        fit = np.dot(design_matrix, coeffs)
+        residuals = fit - y
+        rss = np.sum(residuals**2)
+
+        intercept, alpha, *others = coeffs
+        beta, gamma = np.reshape(others, (2, n_breakpoints))
+
+        cov = (rss / (n_samples - n_coeffs)) * xtx
+
+        return intercept, alpha, beta, gamma, cov, rss
+
+    exitflag = "max iterations reached"
+    breakpoints = np.sort(np.random.uniform(np.min(x), np.max(x), n_breakpoints))
+
+    for _ in range(max_iter):
+        intercept, alpha, beta, gamma, cov, rss = fit_iteration(breakpoints)
+
+        if np.all(gamma < tol):
+            exitflag = "converged"
+            break
+
+        # update breakpoints and check in bounds
+        breakpoints = gamma / beta + breakpoints
```

Comment:
Any reason why we're not just doing:
```suggestion
        breakpoints += gamma / beta 
```